cloudbuild.yaml

```diff
@@ -13,12 +13,214 @@
 # limitations under the License.
 
 
+# This is the Cloud Build build configuration file. It builds and submits the
+# Docker image to Container Registry, and also applies Terraform configurations
+# for the gcloud project based on the current GitHub branch.
+# For more information about Cloud Build, you can find the documentation here:
+# https://cloud.google.com/cloud-build/docs
+# For more information about Terraform, you can find the documentation here:
+# https://www.terraform.io/docs/index.html
+#
+# If Cloud Build is run from a git branch named either "dev" or "prod", 
+# the configuration will be applied to the cloud project provided via
+# environment variable $PROJECT_ID. Otherwise, the changes will be planned,
+# but not applied (dry run mode, which is useful when testing config changes
+# from a development branch).
+#
+# For dev and prod branches, the following steps are executed:
+# 1. terraform init
+# 2. terraform refresh
+# 3. terraform plan
+# 4. terraform apply
+#
+# For any other branch, the following steps are executed:
+# 1. terraform init for all environments subfolders
+# 2. terraform refresh for all environments subfolders
+# 3. terraform plan for all environment subfolders
+#
+# This is to ensure that changes being proposed hold for every single environment.
+# 
+# To test a build on dev manually:
+# 1. Make sure you are using the correct glcoud project for your dev environment:
+#    gcloud config set project PROJECT_ID
+# 2. Make sure you are on the dev branch:
+#    git checkout dev
+# 3. Build and deploy the project
+#    cd ~/cloud-monitoring-notification-delivery-integration-sample-code
+#    gcloud builds submit --config cloudbuild.yaml .
+#
+# To test a build on prod manually:
+# 1. Make sure you are using the correct glcoud project for your prod environment:
+#    gcloud config set project PROJECT_ID
+# 2. Make sure you are on the prod branch:
+#    git checkout prod
+# 3. Build and deploy the project
+#    cd ~/cloud-monitoring-notification-delivery-integration-sample-code
+#    gcloud builds submit --config cloudbuild.yaml .
+
 steps:
 - id: 'build docker image'
-# Uses the docker build step to build an image called cloud-run-pubsub-service
+
+  # Uses the docker build step to build an image
   name: 'gcr.io/cloud-builders/docker'
-  args: ['build', '-t', 'gcr.io/$PROJECT_ID/cloud-run-pubsub-service', './philips_hue_integration_example']
+  args: ['build', '--tag', 'gcr.io/$PROJECT_ID/${_IMAGE_NAME}', './philips_hue_integration_example']
+  timeout: 300s # 5 minutes
+
+- id: 'check branch and project'
+  name: 'bash'
+  args: 
+  - '-c'
+  - | 
+      # check if project id is set
```

Comment:
Super small nit: It might be clearer to factor this step out into a prior bash step, named something like "Check for required environment variables" or the like. Otherwise, LGTM

Reply:
Done